src/CalcManager/CEngine/scicomm.cpp

```diff
@@ -761,12 +766,28 @@ void CCalcEngine::ResolveHighestPrecedenceOperation()
         {
```

Comment:
Another side-effect I'm seeing of this change is that we'll still repeat the user's last action even if they deleted/cleared their history.  That behavior was undefined in the discussion within #264.  @grochocki, do you have thoughts on this?  I'm inclined to think that if the user cleared their history (for whatever reason it was that they had to do so), then they're not going to want their previous action to potentially be exposed by someone hitting <kbd>=</kbd>.  It feels like the ability to do this should be based on history being there.

That further opens the question, if the user deletes the most recent history entry, should hitting <kbd>=</kbd> repeat the _actual_ last action that they had, or should it repeat the last action that their _history_ says that they performed?

Reply:
Interesting question! For information, the current version of Calculator already behaves this way with binary operators:
- Type <kbd>3</kbd> <kbd>+</kbd> <kbd>2</kbd> <kbd>=</kbd>
- Click on `Clear all history` button (the trash icon)
- press <kbd>=</kbd>

In my opinion (for what it's worth), the history is a separate feature and should not change the behavior of the Calculator part but I also understand the possible confusion for users.

Reply:
Another questionable behavior (public version of the app):
- Type <kbd>21</kbd>
- Press <kbd>x²</kbd>
- Press <kbd>CE</kbd>

-> sqrt(21) is added to the History

Reply:
You're right that this is only perpetuating the existing disconnect between history and the repeat-action nature of <kbd>=</kbd>.  Given that, we can table this discussion in a separate Issue and can address it (if we choose to do so) in a separate PR.